internal/cache/data/file_info.go

```diff
@@ -0,0 +1,26 @@
+// Copyright 2023 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package data
+
+type FileInfo struct {
+	Name             string
```

Comment:
should we create another struct for Name and expose a method that gives complete name (appending bucket, bucket creation, object name) to be used as key for storing in File Cache ?

Reply:
``` go
type FileInfoKey struct {
	BucketName string
	BucketCreationTime string
	ObjectName string
}

func (fik FileInfoKey) Key() string {
	return fik.BucketName + fik.BucketCreationTime + fik.ObjectName
}

type FileInfo struct {
	Key             string
	ObjectGeneration string
	Offset           uint64
	FileSize         uint64
}

```

Reply:
Key in FileInfo should be of type FileInfoKey.